mmengine/dist/dist.py

```diff
@@ -308,7 +308,7 @@ def broadcast(data: Tensor,
         backend_device = get_comm_device(group)
         data_on_device = cast_data_device(data, backend_device)
 
-        torch_dist.broadcast(data_on_device, src, group)
+        torch_dist.broadcast(data_on_device.contiguous(), src, group)
```

Comment:
```suggestion
        data_on_device = data_on_device.contiguous()
        torch_dist.broadcast(data_on_device, src, group)
```
`.contiguous` will produce a temporary variable when the tensor is not contiguous. However, broadcasting a temporary variable is of no use. Should first make `data_on_device` contiguous and then broadcast itself.

Reply:
Actually passing a “temporary variable” worked. I have tested this workaround.

Reply:
Thanks for your example. I will modify my PR